tests/platform_tests/api/test_fan.py

```diff
@@ -37,23 +37,57 @@
 STATUS_LED_COLOR_RED = "red"
 STATUS_LED_COLOR_OFF = "off"
 
-
+@pytest.fixture(scope="class")
+def gather_facts(request, duthost):
+    # Get fan truths from platform.json file
+    chassis_truth = duthost.facts.get("chassis")
+    if chassis_truth:
+        request.cls.fan_truth = chassis_truth.get('fans', None)
+        if not request.cls.fan_truth:
+            logger.warning("Unable to get fan_truth from platform.json, test results will not be comprehensive")
```

Comment:
If we want to move this fixture to the PlatformApiTestBase iin the future, then it should probably only obtain the chassis facts, like is being done in test_chassis.py. Then, in your `compare_value_with_platform_facts()` method below, you should get the fan facts from the chassis facts. This way the fixture will be the same for all classes, and will be easy to move to PlatformApiTestBase.

Reply:
fixed